numba/np/numpy_support.py

```diff
@@ -189,10 +193,6 @@ def _check_struct_alignment(rec, fields):
                 raise ValueError(msg.format(npy_align, llvm_align, dt))
 
 
-def is_arrayscalar(val):
-    return np.dtype(type(val)) in FROM_DTYPE
-
-
```

Comment:
This wasn't really a useful function - seeminly it either meant:

* Actually an array scalar type, `isinstance(val, np.scalar)`
* Can be coerced into a array, which just covers all types (`SomeRandomType` is treated as `np.object`).

Reply:
I agree with your assessment of the function purpose. Seems like this is only used internally anyway so its removal is fine if required (I had a quick check on github search, and also looked in https://github.com/numba/numba/issues/5214#issuecomment-593611360).